boudicca.base/common-model/src/test/kotlin/base/boudicca/model/StructuredEventDSLTest.kt

```diff
@@ -0,0 +1,39 @@
+package base.boudicca.model
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import base.boudicca.SemanticKeys
+import base.boudicca.model.structured.*
+import org.junit.jupiter.api.Test
+import java.time.OffsetDateTime
+
+class StructuredEventDSLTest {
+
+    @Test
+    fun `event dsl can be used to create structured event`() {
+
+        val event = structuredEvent("testEvent", OffsetDateTime.now()) {
+            withData(SemanticKeys.DESCRIPTION) {
+                variant(
+                    lang("en"),
+                    markdownFormat(),
+                    data = "# mydata"
+                )
+                variant(
+                    lang("de"),
+                    data = "MEINE DATEN!"
+                )
+            }
+            withData(SemanticKeys.SOURCES) {
+                variant(
+                    listFormat(),
+                    data = "[asdf.com, bsdf.com, csdf.com]"
```

Comment:
this is why i really want typed setters for this. this value will not result in the list think it does (pretty sure at least)

Reply:
will be fixed